### PR TITLE
state tests TestSuite class

### PR DESCRIPTION
**Pull request by winsvega**

### CODE REVIEW

**Comment by gumb0** [comment]
Please rebase, AppVeyor build shoud be fixed in develop now

**Comment by winsvega** [comment]
ok. but do you have any notes so I could change something in PR with rebase?

File: test/tools/jsontests/StateTests.cpp [hunk 1]
```diff
@@ -40,82 +41,99 @@ namespace fs = boost::filesystem;
...
+				if (Options::get().statediff)
+					importer.traceStateDiff();
+			}
 		}
+		return v;
+	}
+
+	std::string suiteFolder() const override
+	{
+		return "GeneralStateTests";
```
Comment (gumb0) [inline test/tools/jsontests/StateTests.cpp]: use `c_GeneralStateTests ` here not to duplicate the string

**Comment by gumb0** [comment]
only one right now :)

File: test/tools/jsontests/StateTests.cpp [hunk 2]
```diff
@@ -40,82 +41,99 @@ namespace fs = boost::filesystem;
...
 
-			//check post hashes against cpp client on all networks
-			mObject post = inputTest.at("post").get_obj();
-			vector<size_t> wrongTransactionsIndexes;
-			for (mObject::const_iterator i = post.begin(); i != post.end(); ++i)
+			if (_fillin)
 			{
-				BOOST_REQUIRE_MESSAGE(i->second.type() == array_type, testname + " post field should contain an array for each network.");
-				for (auto const& exp: i->second.get_array())
+	#if ETH_FATDB
```
Comment (gumb0) [inline test/tools/jsontests/StateTests.cpp]: `#if`, `#else`, `#endif` shouldn't be indented

**Comment by winsvega** [comment]
wait you could push to my repo?

**Comment by gumb0** [comment]
Yeah I don't know, that suprised me, too 😆

**Comment by chfast** [comment]
You can if the author mark "allow maintainers to modify this PR" on GitHub (see right pane).

**Comment by chfast** [comment]
Please wait for CI builds or cancel them if you don't need them.

**Pull request merged**